sdk/src/Util.cs

```diff
@@ -34,7 +34,11 @@ public static string ToJson(IMessage message)
         
         public static string HashAlias(Alias alias)
         {
-            return Base58.Encode(Sha256Hash(Encoding.UTF8.GetBytes(ToJson(alias))));
+            var realm = alias.Realm;
+            alias.Realm = "";
+            var encoded = Base58.Encode(Sha256Hash(Encoding.UTF8.GetBytes(ToJson(alias))));
```

Comment:
It is probably a better practice to create a new Alias, instead of modifying the input one, which is not thread-safe.